python/ray/autoscaler/local/example-full.yaml

```diff
@@ -31,7 +32,8 @@ provider:
 # How Ray will authenticate with newly launched nodes.
 auth:
     ssh_user: YOUR_USERNAME
-    # Optional if an ssh private key is necessary to ssh to the cluster.
+    # If you're starting `ray up example-full.yaml` from a worker node (that has SSH access to all the nodes), this
+    # should be commented out. Only needed if a ssh private key is necessary to ssh to the cluster.
```

Comment:
Would suggest this wording:

The following line is only needed if an ssh private key is necessary to ssh to the Ray cluster.
You can comment out `ssh_private_key` if the following machines don't need a private key for SSH access to the Ray Cluster:
(1) The machine on which `ray up` is executed.
(2) The head node of the Ray cluster.

Reply:
Are you sure this statements are correct (SSH requirements are super fuzzy in the docs..., so let me know if I'm not understanding something)? From https://github.com/ray-project/ray/issues/18529 I'm not sure if (1) is correct since in the stated issue worker1 needs SSH key (for accessing the head node), but workers come up only after commenting this option out.
Regarding (2) do I understand correctly that the head node must have SSH access to all worker nodes? If so how is this configured in the yaml file since this option is only for the e.g. laptop from which the cluster is started?

Reply:
The machine that runs `ray up` executes SSH commands to set up the Ray head node.
The Ray head node subsequently executes SSH commands to set up the Ray worker nodes.

When you run `ray up`, ssh credentials sitting on the `ray up` machine are copied to the head node -- internally, the ssh key is added to the list of file mounts to rsync to head node.

I agree that the docs are bad.

Reply:
I'd really like to add your first two paragraphs to the example file since it clears much of the confusion. But still it's not clear why workers would not be set up if this option remains. Any ideas (so we can also add this to the comments in the example yaml file)?

Reply:
> I'd really like to add your first two paragraphs to the example file

Sounds good.

> why workers would not be set up if this option remains

Sorry, didn't quite follow -- could you clarify?

Reply:
> why workers would not be set up if this option remains

If this option remains (ssh_private_key) in the yaml file, only the head node is provisioned, workers aren't touched. (ray up is started from a worker node that has SSH access to all other nodes). Only with this option commented out, the cluster spins up with all the worker nodes.

Reply:
@DmitriGekhtman  if you don't agree and are happy with the changes, you can just merge the PR. I'm new at using Ray so I might not be right but I think the proposed changes will help first time users like myself.